docs/source/cpp/env_vars.rst

```diff
@@ -132,11 +132,15 @@ that changing their value later will have an effect.
 
 .. envvar:: ARROW_USER_SIMD_LEVEL
 
-   The SIMD optimization level to select.  By default, Arrow C++ detects
-   the capabilities of the current CPU at runtime and chooses the best
-   execution paths based on that information.  One can override the detection
-   by setting this environment variable to a well-defined value.
-   Supported values are:
+   The maximum SIMD optimization level selectable at runtime.  Useful for
+   comparing the performance impact of enabling or disabling respective code
+   paths or working around situations where instructions are supported but are
+   either buggy or not performant.
```

Comment:
Just curious, what does `buggy` here means? ( I'm not familiar with SIMD ). Would that's the case that the arch don't implement the instr well? Or just don't have this instr, do user need disable it?

Reply:
Thanks for taking a look. There may be a better word here...By buggy I was referring to the comment in https://github.com/apache/arrow/issues/40333#issuecomment-1977142577 mentioning situations where like running under a hypervisor where cpu detection reports that an instruction is supported but it's not handled correctly by the hypervisor. I'm not familiar with this so I don't know what "not handled correctly" really means.

Maybe instead of buggy, say "where instructions are supported but are not performant or cause other issues"?

Reply:
Looks nice, we can wait for @pitrou and others' idea here

Reply:
I agree with "where instructions are supported but are not performant or cause other issues"?

Reply:
Thanks! Change made in 5ad7282f7bc3581825098d3789fa0af0c3f0af5f.